src/api/app/controllers/webui/request_controller.rb

```diff
@@ -204,9 +205,14 @@ def sourcediff
   end
 
   def changerequest
-    changestate = (%w[accepted declined revoked new] & params.keys).last
+    changestate = (%w[accepted commented declined revoked new] & params.keys).last
 
-    if change_state(changestate, params)
+    if changestate == 'commented'
+
+      build_new_comment(BsRequest.find_by(number: params[:number]),
```

Comment:
This can be nil

Reply:
We can reuse the `@bs_request` set by `require_request`. If the request is not found it won't even reach this part of the code.